File: core/src/main/kotlin/br/com/dillmann/nginxignition/core/accesslist/AccessList.kt
package br.com.dillmann.nginxignition.core.accesslist

import java.util.UUID

data class AccessList(
    val id: UUID,
    val name: String,
    val realm: String?,
    val defaultOutcome: Outcome,
    val entries: List<EntrySet>,
    val forwardAuthenticationHeader: Boolean,
    val credentials: List<Credentials>,
) {
    enum class Outcome {
        DENY,
        ALLOW,
    }

    data class EntrySet(
        val priority: Int,
        val outcome: Outcome,
        val sourceAddresses: List<String>,
    )

    data class Credentials(
        val username: String,
        val password: String,
    )
}

File: core/access_list/access_list.go
package access_list

import "github.com/google/uuid"

type AccessListOutcome string

const (
	AllowAccessOutcome = AccessListOutcome("ALLOW")
	DenyAccessOutcome  = AccessListOutcome("DENY")
)

type AccessList struct {
	Id                          uuid.UUID
	Name                        string
	Realm                       string
	SatisfyAll                  bool
	DefaultOutcome              AccessListOutcome
	Entries                     []AccessListEntry
	Credentials                 []AccessListCredentials
	ForwardAuthenticationHeader bool
}

type AccessListEntry struct {
	Priority      int64
	Outcome       AccessListOutcome
	SourceAddress []string
}

type AccessListCredentials struct {
	Username string
	Password string
}
